lib/preload.js

```diff
@@ -1,32 +1,125 @@
 'use strict'
 
-const childProcess = require('child_process')
 const path = require('path')
-const escape = require('command-join')
-const compareVersions = require('compare-versions')
+const Docker = require('dockerode')
+const tarfs = require('tar-fs')
```

Comment:
I would watch out for using this module. The underlying `tar-stream` module has a bug when using the streaming interface which can cause hanging. To solve this it's best to use `tar-stream` directly, with the `Buffer` interface (but it is less efficient).